lxd/instance/drivers/driver_qemu.go

```diff
@@ -4686,6 +4689,43 @@ func (d *qemu) Stop(stateful bool) error {
 		return fmt.Errorf("Stateful stop requires migration.stateful to be set to true")
 	}
 
+	pool, err := d.getStoragePool()
+	if err != nil {
+		return err
+	}
+
+	sharedStorage := pool.Driver().Info().Remote
+	if sharedStorage {
+		_, rootDiskDevice, err := d.getRootDiskDevice()
+		if err != nil {
+			return err
+		}
+
+		stateDiskSizeStr := pool.Driver().Info().DefaultVMBlockFilesystemSize
```

Comment:
This logic is a duplicate of the logic in that start function.
We should separate it into a shared function on the qemu struct called `d.validateRootDiskStatefulStop()` and use in both.

Reply:
We should do this check for all storage drivers not just remote ones as discussed the other day, that way it will be ready for when we do 3.

>Investigate if we can make size.state be specified on a running VM, as its just growing the agent config volume and not the actual VM root volume. If so then it will be possible to enable a VM using local storage for live migration/stateful stop without stopping it, and will allow VMs running on shared storage to be enabled for stateful stop without stopping it first.